### PR TITLE
mgr/dashboard: Remove _filterValue from CdFormGroup

### PR DESCRIPTION
**Pull request by Devp00l**
The current behavior when submitting an empty mail field in the RGW user form
that was previously filled was to replace it with "false".

This was a regression introduced through the use of "CdFormGroup" which had
a "_filterValue" method which converted empty strings into false, but it was
removed now.

Fixes: http://tracker.ceph.com/issues/26861
Signed-off-by: Stephan Müller <smueller@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts [hunk 1]
```diff
@@ -113,6 +113,7 @@ export class RgwUserFormComponent implements OnInit {
         ]
       ]
     });
+    this.userForm._filterValue = (value) => value;
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: Please add a comment here why it is necessary to override this function.

**Comment by votdev** [comment]
Can you please add a comment that explains the need of the ``_filterValue`` function (https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-group.ts#L63)? After reviewing the code of the [getValue function](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-group.ts#L57) the intention is still not clear to me.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts (hunk 1 above)
Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: IMHO, a comment on that function doesn't make the implementation better. What we really need, and that's what you've already said @votdev, is a refactoring.

**Comment by Devp00l** [comment]
jenkins test dashboard

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by Devp00l** [comment]
Addressed all comments

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.spec.ts [hunk 2]
```diff
@@ -144,4 +141,23 @@ describe('RgwUserFormComponent', () => {
       })
     );
   });
+
+  describe('onSubmit', () => {
+    it('should be able to clear the mail field on update', () => {
+      spyOn(rgwUserService, 'update');
+      component.editing = true;
+
+      const user_id = component.userForm.get('email');
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.spec.ts]: s/user_id/email/

**Comment by Devp00l** [comment]
jenkins retest this please

**Pull request merged**